src/modalities/dataloader/create_packed_data.py

```diff
@@ -175,3 +176,77 @@ def _process_line(self, line: str) -> bytes:
         if len(tokens) == 0:
             raise EmptySampleError("Received empty sample...")
         return b"".join(map(self._encoded_token_to_bytes, tokens)) + self._encoded_eos_token_as_bytes
+
+
+class EmbeddedStreamData:
+    # amount of bytes to represent number of all tokens in dataset.
+    # If the amount exceeds 2^(8*`header_size_in_bytes`), this requires adaptation.
+    # Decided to keep this constant, since a size of 8 bytes requires more data than the internet currently provides
+    DATA_SECTION_LENGTH_IN_BYTES = 8
+    TOKEN_SIZE_DESCRIPTOR_LENGTH_IN_BYTES = 4
+    HEADER_SIZE_IN_BYTES = DATA_SECTION_LENGTH_IN_BYTES + TOKEN_SIZE_DESCRIPTOR_LENGTH_IN_BYTES
+
+    def __init__(self, data_path: Path):
+        self._data_path = data_path
+        if not self._data_path.is_file():
+            raise FileNotFoundError(
```

Comment:
I'd agree, since this error message is more detailed than just mentioning a missing directory or file. However these partially different functionalities. The checks of types [here](https://github.com/Modalities/modalities/pull/48/files#diff-e818bc80be6213876645f342dd913db6a5fcb7228f31f58877c5fa454def634eR200) are done to collect all ".pbin" files into a list to be merged. So checking for directories is a must. However we could skip the file check and the `ValueError`.

I will make a respective proposition

Reply:
please have another look at https://github.com/Modalities/modalities/pull/48/commits/b0be92952c753030c05254322796d3ecc20382d5